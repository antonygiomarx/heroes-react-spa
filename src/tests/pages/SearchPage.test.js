const { mount } = require("enzyme");
const { MemoryRouter, Route } = require("react-router-dom");
const { SearchPage } = require("../../pages/Search/SearchPage");

describe('Test in Search Page', () => {
    
    test('should show itself correctly', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path="/search" component={ SearchPage } />
            </MemoryRouter>
        );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.alert__info').exists() ).toBe(true)

    });

    test('should show the input value equal to superman', () => {

        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path="/search" component={ SearchPage } />
            </MemoryRouter>
        );
        let input = wrapper.find('input').prop('value')
        input = 'batman'
        expect( input ).toBe('batman');
        
    });

    test('should show a error if the heroe is not finded', () => {
        
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <Route path="/search" component={ SearchPage } />
            </MemoryRouter>
        );

        expect( wrapper.find('.alert__error').text().trim() ).toBe(`The heroe  batman123  do not exist`);
        expect( wrapper ).toMatchSnapshot();

    });
    
    test('should call the history push', () => {
        
        const history = {
            push: jest.fn()
        };

        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <Route 
                    path="/search" 
                    component={ () => <SearchPage history={ history } /> } 
                />
            </MemoryRouter>
        );

        wrapper.find('input').simulate('change', {
            target: {
                name: 'searchText',
                value: 'batman'
            }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        })

        expect( history.push ).toHaveBeenCalledWith(`?q=batman`);
        
    });
    
   

});