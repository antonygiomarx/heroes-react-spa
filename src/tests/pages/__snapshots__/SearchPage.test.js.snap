// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Search Page should show a error if the heroe is not finded 1`] = `
<main
  className="container"
>
  <header
    className="HeaderPage"
  >
    <h1
      className="header-page__content"
    >
       
      Search
       
    </h1>
  </header>
  <section
    className="search"
  >
    <div
      className="search__form"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          name="searchText"
          onChange={[Function]}
          placeholder="Search a hero"
          type="text"
          vakue=""
        />
        <button
          className="btn__search"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="search__results"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert__error alert"
      >
        The heroe 
        <strong>
           
          batman123
           
        </strong>
         do not exist
      </div>
    </div>
  </section>
</main>
`;

exports[`Test in Search Page should show itself correctly 1`] = `
<main
  className="container"
>
  <header
    className="HeaderPage"
  >
    <h1
      className="header-page__content"
    >
       
      Search
       
    </h1>
  </header>
  <section
    className="search"
  >
    <div
      className="search__form"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          name="searchText"
          onChange={[Function]}
          placeholder="Search a hero"
          type="text"
          vakue=""
        />
        <button
          className="btn__search"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="search__results"
    >
      <h4>
         Results 
      </h4>
      <hr />
      <div
        className="alert__info alert"
      >
        You most search a heroe
      </div>
    </div>
  </section>
</main>
`;
